src/prompts.ts

```diff
@@ -73,339 +78,378 @@ export function readProviderPromptMap(
+  if (typeof promptPathOrGlobs === 'object' && Object.keys(promptPathOrGlobs).length) {
+    /* NOTE: This format is considered legacy and has been deprecated. Example:
+    {
+      'prompts.txt': 'foo1',
+      'prompts.py': 'foo2',
+    }
+    */
+    return Object.entries(promptPathOrGlobs).map(([raw, key]) => ({
+      label: key,
+      raw: raw,
+      source: raw,
+    }));
   }
-  return promptContents;
+  // numbers, booleans, etc
+  throw new Error(`Invalid input prompt: ${JSON.stringify(promptPathOrGlobs)}`);
 }
 
-export const DEFAULT_GRADING_PROMPT = JSON.stringify([
-  {
-    role: 'system',
-    content: `You are grading output according to a user-specified rubric. If the statement in the rubric is true, then the output passes the test. You respond with a JSON object with this structure: {pass: boolean; reason: string;}.
-
-Examples:
+/**
+ * Processes a JSONL file to extract prompts.
+ * @param {string} filePath - Path to the JSONL file.
+ * @param {string | undefined} labelPrefix - Optional prefix for labels.
+ * @param {string} rawPath - Raw path of the file.
+ * @returns {Prompt[]} Array of prompts extracted from the file.
+ */
+function processJsonlFile(
+  filePath: string,
+  labelPrefix: string | undefined,
+  rawPath: string,
+): Prompt[] {
+  const fileContent = fs.readFileSync(filePath, 'utf-8');
+  const jsonLines = fileContent.split(/\r?\n/).filter((line) => line.length > 0);
+  return jsonLines.map((json) => ({
+    raw: json,
+    label: `${labelPrefix ? `${labelPrefix}: ` : ''}${rawPath}: ${json}`,
+  }));
+}
 
-Output: Hello world
-Rubric: Content contains a greeting
-{"pass": true, "score": 1.0, "reason": "the content contains the word 'world'"}
+/**
+ * Processes a text file to extract prompts, splitting by a delimiter.
+ * @param {string} promptPath - Path to the text file.
+ * @param {RawPrompt} prompt - The raw prompt data.
+ * @returns {Prompt[]} Array of prompts extracted from the file.
+ */
+function processTxtFile(promptPath: string, prompt: RawPrompt): Prompt[] {
+  const fileContent = fs.readFileSync(promptPath, 'utf-8');
+  return fileContent
+    .split(PROMPT_DELIMITER)
+    .map((p) => ({
+      raw: p.trim(),
+      label: prompt.label
+        ? `${prompt.label}: ${prompt.raw}: ${p.trim()}`
+        : `${prompt.raw}: ${p.trim()}`,
+    }))
+    .filter((p) => p.raw.length > 0); // handle leading/trailing delimiters and empty lines
+}
 
-Output: Avast ye swabs, repel the invaders!
-Rubric: Does not speak like a pirate
-{"pass": false, "score": 0.0, "reason": "'avast ye' is a common pirate term"}`,
-  },
-  {
-    role: 'user',
-    content: 'Output: {{ output }}\nRubric: {{ rubric }}',
+/**
+ * Python prompt function. Runs a specific function from the python file.
+ * @param promptPath - Path to the Python file.
+ * @param functionName - Function name to execute.
+ * @param context - Context for the prompt.
+ * @returns The prompts
+ */
+export const promptFunction = async (
```

Comment:
let's name `promptFunction` and `promptFunctionLegacy` to indicate that they are for python